### ISSUE TITLE
CRAN package - scholar

### ISSUE THREAD

**Issue by GuangchuangYu** [ropensci/software-review#234]
### Summary

-   What does this package do? (explain in 50 words or less):

The package provides functions to extract citation data and co-author network from Google Scholar. 

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: scholar
Authors@R: c(
    person("Guangchuang", "Yu",   email = "guangchuangyu@gmail.com",   role = c("aut", "cre"), comment = c(ORCID = "0000-0002-6485-8781")),
    person("James", "Keirstead",  email = "james.keirstead@gmail.com", role = "aut"),
    person("Gregory", "Jefferis", email = "jefferis@gmail.com",        role = "ctb"),
    person("Jorge", "Cimentada", email = "cimentadaj@gmail.com",        role = "ctb"),
    person("Max", "Czapanskiy", email = "maxczapanskiy@gmail.com",        role = "ctb")
    )
Maintainer: Guangchuang Yu <guangchuangyu@gmail.com>
Version: 0.1.7
License: MIT + file LICENSE
Title: Analyse Citation Data from Google Scholar
BugReports: https://github.com/jkeirstead/scholar/issues
Description: Provides functions to extract citation data from Google
    Scholar.  Convenience functions are also provided for comparing
    multiple scholars and predicting future h-index values.
LazyData: true
Depends:
    R (>= 3.4.0)
Imports:
    R.cache,
    dplyr,
    httr,
    rvest,
    stringr,
    xml2,
    tidygraph,
    ggraph,
    ggplot2
Suggests:
    knitr,
    prettydoc,
    roxygen2
VignetteBuilder: knitr
RoxygenNote: 6.0.1

```

-   URL for the package (the development repository, not a stylized html page):

<https://github.com/jkeirstead/scholar>

- Please indicate which category or categories from our [package fit policies](https://github.com/ropensci/onboarding/blob/master/policies.md#package-fit) this package falls under ***and why**(? (e.g., data retrieval, reproducibility. If you are unsure, we suggest you make a pre-submission inquiry.):

[e.g., "data extraction, because the package parses a scientific data file format"]

data retrieval, because the package extract data from online service.

-   Who is the target audience and what are scientific applications of this package?  

Users who want to analyze citation data.

-   Are there other R packages that accomplish the same thing? If so, how does
yours differ or meet [our criteria for best-in-category](https://github.com/ropensci/onboarding/blob/master/policies.md#overlap)?

No (to my knowledge).

-   If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

### Requirements

Confirm each of the following by checking the box.  This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] has a CRAN and OSI accepted license.
- [x] contains a README with instructions for installing the development version. 
- [x] includes documentation with examples for all functions.
- [ x contains a vignette with examples of its essential functions and uses.
- [ ] has a test suite.
- [ ] has continuous integration, including reporting of test coverage, using services such as Travis CI, Coveralls and/or CodeCov.
- [x] I agree to abide by [ROpenSci's Code of Conduct](https://github.com/ropensci/onboarding/blob/master/policies.md#code-of-conduct) during the review process and in maintaining my package should it be accepted.

#### Publication options

- [x] Do you intend for this package to go on CRAN?  (already on CRAN)
- [ ] Do you wish to automatically submit to the [Journal of Open Source Software](http://joss.theoj.org/)? If so:
    - [ ] The package has an **obvious research application** according to [JOSS's definition](http://joss.theoj.org/about#submission_requirements).
    - [ ] The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with a high-level description in the package root or in `inst/`.
    - [ ] The package is deposited in a long-term repository with the DOI: 
    - (*Do not submit your package separately to JOSS*)
- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:
    - [ ] The package is novel and will be of interest to the broad readership of the journal. 
    - [ ] The manuscript describing the package is no longer than 3000 words.
    - [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
    - (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no gaurantee that your manuscript willl be within MEE scope.*)
    - (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
    - (*Please do not submit your package separately to Methods in Ecology and Evolution*)

### Detail

- [x] Does `R CMD check` (or `devtools::check()`) succeed?  Paste and describe any errors or warnings:

- [x] Does the package conform to [rOpenSci packaging guidelines](https://github.com/ropensci/onboarding/blob/master/packaging_guide.md)? Please describe any exceptions:

- If this is a resubmission following rejection, please explain the change in circumstances:

- If possible, please provide recommendations of reviewers - those with experience with similar packages and/or likely users of your package - and their GitHub user names:

**Comment by GuangchuangYu** [comment]
@jkeirstead is the original author that created the package. James's work is out of academia and have no time to further develop/maintain the package.

I am the new maintainer of the package. However, I should be admitted that I am not the expert of web technologies.

As this package was contributed by several developers, including @cimentadaj @DominiqueMakowski @abfleishman @dfalster and @ jefferis, it would be great to move the package to ROpenSci and maintained by the community.

**Comment by maelle** [comment]
Thanks for your submission @guangchuangyu! Just to clarify, do you intend to keep maintaining the package if it is onboarded, or are you looking for a new maintainer?

**Comment by GuangchuangYu** [comment]
@maelle both are fine to me :)

**Comment by maelle** [comment]
Well for onboarding it'd be great if you remained maintainer, and you could look for a new maintainer later if/when you realize you don't have time any more and there's a good candidate. Does that sound good? Otherwise the maintainer change would have to happen before reviews, not during the onboarding process. 😸

**Comment by maelle** [comment]
Also note that the package will need some unit tests and continuous integration before the submission process starts. More details [here](https://ropensci.github.io/dev_guide/building.html). We can put this submission on hold while you add them. 😺

**Comment by GuangchuangYu** [comment]
thanks @maelle. 

Sounds good and will add unit tests soon.

**Comment by maelle** [comment]
@jefferis :wave:, @sckott can probably tell you more, since he maintains `vcr` and `webmockr` and has a [WIP book about http testing with `crul`](https://github.com/ropensci/http-testing-book). We don't have guidance about API package testing in our gitbook yet, but there was [a thread about it in our forum](https://discuss.ropensci.org/t/best-practices-for-testing-api-packages/460).

**Comment by jefferis** [comment]
Thanks @maelle. My conclusion from looking at those is that we either:
* switch to crul and use vcr
* roll our own
* use https://cran.r-project.org/web/packages/httptest/index.html
@sckott any additional comments?

**Comment by sckott** [comment]
switching to `crul` is the easiest option if you want mocking and/or caching of real requests/responses. `crul` has some other benefits in that it has a pagination helper in case you need to do that, and has async built in if you need to do that. I've got a number of packages on CRAN already using `vcr` - see the reverse suggests here https://cran.rstudio.com/web/packages/vcr/  

I am working on adapters in `webmockr` and `vcr` for both `curl` and `httr`, but they aren't quite ready yet. 

I've not tried httptest or other caching options, so can't comment on those

**Comment by annakrystalli** [comment]
Hello @GuangchuangYu 👋 

I'll be handling your review once you're ready to submit properly. 

Ping me on here when you're all set up with your tests and I'll move forward with the initial editor's checks. Feel free to reach out with any questions in the mean time.

**Comment by GuangchuangYu** [comment]
thanks @annakrystalli .

Will get in touch when everything was ready (so much things need to learn).

**Comment by annakrystalli** [comment]
Hello @GuangchuangYu. Just checking on this submission and to remind you that issues on hold are closed after 1 year of inactivity. That still gives you till 3rd of July 2019 but wanted to make you aware. As always feel free to reach out with any questions. ☺️

**Comment by annakrystalli** [comment]
Hi @GuangchuangYu. As there has been no activity on this issue for over a year I will be closing it. Let us know if you wish to resubmit in the future once it is ready.
